src/NationalInstruments.Analyzers/Correctness/RecordWithEnumerablesShouldOverrideDefaultEqualityAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Correctness
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class RecordWithEnumerablesShouldOverrideDefaultEqualityAnalyzer
+        : NIDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "NI1019";
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            new LocalizableResourceString(nameof(Resources.NI1019_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1019_Message), Resources.ResourceManager, typeof(Resources)),
+            Category.Correctness,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: new LocalizableResourceString(nameof(Resources.NI1019_Description), Resources.ResourceManager, typeof(Resources)));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterSymbolAction(AnalyzeRecord, SymbolKind.NamedType);
+        }
+
+        private void AnalyzeRecord(SymbolAnalysisContext context)
+        {
+            var typeSymbol = (ITypeSymbol)context.Symbol;
+            if (typeSymbol.IsRecord == false)
+            {
+                return;
+            }
+
+            var baseTypeProperties = GetBaseTypeProperties(typeSymbol)
+                .Select(p => p.Name)
+                .ToImmutableHashSet();
+
+            var enumerableProperties = typeSymbol
+                .GetPublicPropertySymbols()
+                .Where(p => p.Type.IsEnumerable() && !baseTypeProperties.Contains(p.Name))
+                .ToImmutableArray();
+
+            if (enumerableProperties.Length == 0)
+            {
+                return;
+            }
+
+            // check if the record type has implemented its own Equality methods
+            foreach (var location in typeSymbol.Locations)
```

Comment:
Rather than check for the Equals method syntax, what about the following:

```cs
// Convert to INamedTypeSymbol vs. TypedSymbol at start
var namedTypeSymbol = (INamedTypeSymbol)context.Symbol;
...
if (namedTypeSymbol.ExplicitlyImplementsEquals())
{
    return;
}
```

Create INamedTypeSymbolExtensions.cs in NationalInstruments.Analyzers.Utilities.Extensions:

```cs
public static class NamedTypeSymbolExtensions
{
    public static bool ExplicitlyImplementsEquals(this INamedTypeSymbol namedTypeSymbol)
        => namedTypeSymbol.GetMembers().OfType<IMethodSymbol>().Any(m => m.IsExplicitEquals());
}
```

Add to IMethodSymbolExtensions.cs:

```cs
public static bool IsExplicitEquals(this IMethodSymbol method)
    => method != null
        && method.Name == WellKnownMemberNames.ObjectEquals
        && method.ReturnType.SpecialType == SpecialType.System_Boolean
        && method.Parameters.Length == 1
        && !method.IsImplicitlyDeclared;
```

Reply:
Sure that's definitely an improvement!  I'll add it.